src/main.ts

```diff
@@ -284,36 +301,70 @@ function getFetchInfo(ref: string, commit: string): IFetchInfo {
   return result
 }
 
-async function configGitAuth(token: string) {
+async function configGitAuthForSubmodules(token: string, repoDir: string) {
+  // Set authentication
+  const basicCredential = Buffer.from(
+    `x-access-token:${token}`,
+    'utf8'
+  ).toString('base64')
+  core.setSecret(basicCredential)
+
+  await exec.exec(
+    `git submodule foreach --recursive sh -c "git config --local --add 'http.https://github.com/.extraheader' 'AUTHORIZATION: basic ${basicCredential}'"`,
```

Comment:
Optional: I guess when we're splitting the command and running two git submoduel each in sequence, we don't need sh -c (it would probably be needed to evaluate a && or || in between commands)